doc/acl/Dynamic-ACL-Design.md

```diff
@@ -0,0 +1,198 @@
+
+# Time-based ACL #
+
+
+## Table of Content
+
+- [Time-based ACL](#time-based-acl)
+  - [Table of Content](#table-of-content)
+  - [Revision](#revision)
+  - [Scope](#scope)
+  - [Definitions/Abbreviations](#definitionsabbreviations)
+  - [Overview](#overview)
+  - [Requirements](#requirements)
+  - [Architecture Design](#architecture-design)
+  - [High-Level Design](#high-level-design)
+    - [db schema](#db-schema)
+      - [CONFIG DB](#config-db)
+      - [APP DB](#app-db)
+      - [acl-loader](#acl-loader)
+      - [dynamic_acl_mgrd](#dynamic_acl_mgrd)
+    - [Work flow](#work-flow)
+      - [Add and remove time-based ACL rule](#add-and-remove-time-based-acl-rule)
+      - [Refresh TTL of existing ACL rule](#refresh-ttl-of-existing-acl-rule)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+      - [System Test cases](#system-test-cases)
+  - [Open questions](#open-questions)
+  -[Work flow](#work-flow)
+    - [Add dynamic ACL rule](#add-dynamic-acl-rule)
+    - [Refresh TTL of existing ACL rule](#refresh-ttl-of-existing-acl-rule)
+    - [Remove ACL rule when TTL expires](#remove-acl-rule-when-ttl-expires)
+  - [Testing Requirements/Design](#testing-requirementsdesign)
+    - [Unit Test cases](#unit-test-cases)
+    - [System Test cases](#system-test-cases)
+- [Open questions](#open-questions)
+
+
+## Revision
+
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 |             | Bing Wang          | Initial version                   |
+| 0.2 |9/6/2022     | Shiyan Wang        |                                   |
+
+## Scope
+
+The scope of this document covers the design of dynamic data plane ACL, which assigns a TTL to ACL rule, and removes the expired rule. Control plane ACL is not covered in this document.
+
+## Definitions/Abbreviations 
+
+| Definitions/Abbreviation | Description                                |
+|--------------------------|--------------------------------------------|
+| ACL                      | Access Control List                        |
+
+
+## Overview 
+
+The current design of data plane ACL supports only persistent rules, that is, once the rule is applied to SONiC, it will be there until manually removed or config reload.
+
+This doc proposes an enhancement to current ACL, which add a TTL to ACL rules and the rules are deleted after TTL expires.
```

Comment:
which add a `time range` to ACL rules and `the rule will only be activated during the time range`

Reply:
Accepted and thanks!